core/src/client/async_client/manager.rs

```diff
@@ -259,10 +261,11 @@ impl RequestManager {
 		subscription_id: SubscriptionId<'static>,
 	) -> Option<(RequestId, SubscriptionSink, UnsubscribeMethod, SubscriptionId)> {
 		match (self.requests.entry(request_id), self.subscriptions.entry(subscription_id)) {
-			(Entry::Occupied(request), Entry::Occupied(subscription))
+			(Entry::Occupied(mut request), Entry::Occupied(subscription))
 				if matches!(request.get(), Kind::Subscription(_)) =>
 			{
-				let (_req_id, kind) = request.remove_entry();
+				// Mark the request ID as pending unsubscription.
+				let kind = std::mem::replace(request.get_mut(), Kind::PendingMethodCall(None));
 				let (sub_id, _req_id) = subscription.remove_entry();
 				if let Kind::Subscription((unsub_req_id, send_back, unsub)) = kind {
```

Comment:
Nit: we could remove the unreachable by doing something like:

```rust
match (self.requests.entry(request_id), self.subscriptions.entry(subscription_id)) {
    (Entry::Occupied(mut request), Entry::Occupied(subscription))
        if matches!(request.get(), Kind::Subscription(_)) =>
    {
        let Kind::Subscription((unsub_req_id, send_back, unsub)) = request.get() else {
            return None
        };
    }
```

Maybe a bit cleaner :)

Reply:
yeah but I think I did in this way because the `None` should be unreachable i.e, a logic bug lemme have a look if I broke something or simply remembering this wrong...

Reply:
ah, the unreachable can be removed done...